lib/middleware/session.js

```diff
@@ -79,7 +79,14 @@ var ServletSession = exports.ServletSession = function(request) {
                 // to the attributes in the servlet session object.
                 data = new JavaAdapter(org.mozilla.javascript.NativeObject, {
                     put: function(name, start, value) {
-                        getSession().setAttribute(name, value);
+                    	if (typeof(value) === 'undefined' || value === null) {
```

Comment:
This conforms to HttpSession's behavior:

> If the value passed in is null, this has the same effect as calling removeAttribute().

http://docs.oracle.com/javaee/7/api/javax/servlet/http/HttpSession.html#setAttribute-java.lang.String-java.lang.Object-